grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaChannel.java

```diff
@@ -149,6 +150,7 @@ public <I, O> ClientCall<I, O> newCall(MethodDescriptor<I, O> method, CallOption
         final DefaultClientRequestContext ctx = newContext(HttpMethod.POST, req, method);
 
         GrpcCallOptions.set(ctx, callOptions);
+        GrpcMethodDescriptor.set(ctx, method);
```

Comment:
I've been hesitant to add another API for retrieving gRPC call contexts. Would it make sense to deprecate `GrpcCallOptions` and instead introduce a unified API to use gRPC context in decorators?
```java
// A public API
public final class GrpcContexts {

    // Expose only getters
    public MethodDescriptor methodDescriptor(RequestContext ctx) {
        return InternalGrpcContexts.methodDescriptor(ctx);
    }

    public CallOptions callOptions(RequestContext ctx) {
        ...
    }
}

// An internal API
public class InternalGrpcContexts {

    MethodDescriptor methodDescriptor(RequestContext ctx);
    void methodDescriptor(RequestContext ctx, MethodDescriptor method);

    ...
}
```